docs/rules-engine/how-to-use/dedicated-component-ruleset.md

```diff
@@ -1,15 +1,60 @@
-# Rules engine
+# Rules engine - Dedicated ruleset for component
 
-## Dedicated ruleset for component
+The goal of the feature is to create a ruleset that will be executed if and only if it's used. This will improve the performances 
+of your application as the rules engine will not be triggered if the facts are updated but your rule has no effect
+at the moment -- for instance if it is supposed to change the configuration of a missing component.
 
-### Description
+To do so, let's introduce the concept of ``linkedComponents``. These are the components that will enable or disable the
+ruleset they are linked to. This is done in their ``ngOnInit`` and ``ngOnDestroy`` methods.
 
-The goal of the feature is to allow to create a ruleset that will be executed if and only if it's used.
-To do that we will subscribe in the onInit of the component, to the dedicated ruleset.
+## How to make a Component compatible
 
-## How to add it to your application
+You can make a component ``linkedComponents`` compatible thanks the following Otter CLI command:
+``ng g rules-engine-to-component --path=[path-to-my-component-file]``
 
-First, you need to have a ruleset defined in your json (created via the CMS interface) that will be identified by a unique id, and you need to specify that the ruleSet is linked to at least one component, meaning that it's not executed per default. The json will look as follows :
+This command will compute the component's identifier and add the methods to enable or disable the ruleset 
+matching the component name.
+
+```typescript
```

Comment:
instead of typescript we could use `diff`
to easily understand what will be modified
```diff
import {ChangeDetectionStrategy, Component, ViewEncapsulation, inject, OnInit, OnDestroy} from '@angular/core';
+ import {LinkableToRuleset, RulesEngineRunnerService} from '@o3r/rules-engine';
+ import {computeItemIdentifier} from '@o3r/core';

@O3rComponent({ componentType: 'Component' })
@Component({
  selector: 'app-my-component',
  templateUrl: './my-component.template.html',
  styleUrl: './my-component.style.scss',
  encapsulation: ViewEncapsulation.None,
  changeDetection: ChangeDetectionStrategy.OnPush
})
- export class MyComponent {
+ export class MyComponent implements OnInit, OnDestroy {

+  private readonly rulesEngineService = inject(RulesEngineRunnerService, { optional: true });
+  private readonly componentIdentifier = computeItemIdentifier('MyComponent', '@scope/components');

+  public ngOnInit() {
+    if (this.rulesEngineService) {
+      this.rulesEngineService.enableRuleSetFor(this.componentName);
+    }
+  }
+  public ngOnDestroy() {
+    if (this.rulesEngineService) {
+      this.rulesEngineService.disableRuleSetFor(this.componentName);
+   }
+  }
}
```